google-beta/resource_compute_network.go

```diff
@@ -75,139 +92,299 @@ func resourceComputeNetwork() *schema.Resource {
 func resourceComputeNetworkCreate(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
 
-	project, err := getProject(d, config)
+	obj := make(map[string]interface{})
+	descriptionProp, err := expandComputeNetworkDescription(d.Get("description"), d, config)
 	if err != nil {
 		return err
+	} else if v, ok := d.GetOkExists("description"); !isEmptyValue(reflect.ValueOf(descriptionProp)) && (ok || !reflect.DeepEqual(v, descriptionProp)) {
+		obj["description"] = descriptionProp
 	}
-
-	//
-	// Possible modes:
-	// - 1 Legacy mode - Create a network in the legacy mode. ipv4_range is set. auto_create_subnetworks must not be
-	//     set (enforced by ConflictsWith schema attribute)
-	// - 2 Distributed Mode - Create a new generation network that supports subnetworks:
-	//   - 2.a - Auto subnet mode - auto_create_subnetworks = true, Google will generate 1 subnetwork per region
-	//   - 2.b - Custom subnet mode - auto_create_subnetworks = false & ipv4_range not set,
-	//
-	autoCreateSubnetworks := d.Get("auto_create_subnetworks").(bool)
-	if autoCreateSubnetworks && d.Get("ipv4_range").(string) != "" {
-		return fmt.Errorf("ipv4_range can't be set if auto_create_subnetworks is true.")
+	IPv4RangeProp, err := expandComputeNetworkIpv4_range(d.Get("ipv4_range"), d, config)
+	if err != nil {
+		return err
+	} else if v, ok := d.GetOkExists("ipv4_range"); !isEmptyValue(reflect.ValueOf(IPv4RangeProp)) && (ok || !reflect.DeepEqual(v, IPv4RangeProp)) {
+		obj["IPv4Range"] = IPv4RangeProp
 	}
-
-	// Build the network parameter
-	network := &compute.Network{
-		Name:                  d.Get("name").(string),
-		AutoCreateSubnetworks: autoCreateSubnetworks,
-		Description:           d.Get("description").(string),
+	nameProp, err := expandComputeNetworkName(d.Get("name"), d, config)
+	if err != nil {
+		return err
+	} else if v, ok := d.GetOkExists("name"); !isEmptyValue(reflect.ValueOf(nameProp)) && (ok || !reflect.DeepEqual(v, nameProp)) {
+		obj["name"] = nameProp
+	}
+	autoCreateSubnetworksProp, err := expandComputeNetworkAutoCreateSubnetworks(d.Get("auto_create_subnetworks"), d, config)
```

Comment:
Looks like we conditionally need a send_empty_value on this, what do you think?

Reply:
Done in the encoder: https://github.com/terraform-providers/terraform-provider-google-beta/pull/509/files#diff-40eee92a42c8c9e966fc384afde6f29fR384

Reply:
+ behaviour verified by a new test. We had no coverage before 🤦‍♀️

Reply:
Haha, awesome.  Thanks!